ephios/plugins/files/models.py

```diff
@@ -31,6 +36,22 @@ def run_on_commit():
     on_commit(run_on_commit, using)
 
 
+def _generate_token():
+    return "".join(
+        random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(32)
+    )
+
+
+def _token_expiration():
+    return timezone.now() + timedelta(hours=1)
+
+
+class DownloadTicket(models.Model):
+    document = models.ForeignKey(Document, on_delete=models.CASCADE)
```

Comment:
Like this the whole setup is not transferable to other files from other models. If we introduce custom logos or profile pics etc we would need to find a different solution or add a new model each. There must be a better way.